openpgp/packet/public_key.go

```diff
@@ -65,18 +65,17 @@ type edDSAkey struct {
 }
 
 func (e *edDSAkey) Verify(payload []byte, r parsedMPI, s parsedMPI) bool {
-	var key [ed25519.PublicKeySize]byte
 	var sig [ed25519.SignatureSize]byte
 
 	// NOTE: The first byte is 0x40 - MPI header
 	// TODO: Maybe clean the code up and use 0x40 as a header when
 	// reading and keep only actual number in p field. Find out how
 	// other MPIs are stored.
-	copy(key[:], e.p.bytes[1:])
+	key := e.p.bytes[1:]
```

Comment:
we should assert here that we `e.p.bytes` is right size, because `ed25519` will just panic if it gets wrong sized buffers IIRC.